CONTRIBUTING.md

```diff
@@ -0,0 +1,63 @@
+# Contributor Onboarding
+
+🚧 This document is still a WIP 🚧 
+
+## General
+
+ - The goal of this guide is to help you start contributing to Durable Functions
+
+## Pre-reqs
+
+ - OS
+    - Windows 10 (suggested)
+    - Mac OS X/Linux
+ - Language runtimes
+    - .NET Core 2.0
+ - Editor
+    - Visual Studio 2017 (recommended)
+    - VS Code 
+ - Misc tools (suggested)
+    - git - source control
+    - [Azure Storage Emulator](https://docs.microsoft.com/azure/storage/storage-use-emulator) or real Azure Storage Account
+
+## Testing code changes locally (Visual Studio) 
+
+Durable Functions are distributed as a NuGet package. So in order to test your changes, you need to integrate it in a function. For that you need the following steps:
+
+1. Create a local NuGet source on your dev machine - e.g. C:\LocalNuGet.
+2. Then add the local NuGet source as a NuGet package source in Visual Studio. 
+3. Modify the code.
+4. Build the project and this will create a new NuGet package with your changes.
+5. Add the newly built Durable Functions NuGet package to the local NuGet source. The command to add packages to this location is nuget **add package.nupkg -Source C:\LocalNuGet** . Add the symbols file as well to be able to debug your modifications while debugging the function code. 
+5. Update your local NuGet cache (**%USERPROFILE%\.nuget\packages**) with the newest version of the Durable Functions NuGet package.
+6. In Visual Studio, add the new NuGet package from your local NuGet source to a function. 
+7. Run Azure Storage Emulator 5.2.
+8. Run the function and debug.
+
+## Running the tests (Visual Studio) 
+
+1. Build the project and Visual Studio will identify all the tests in the solution.
+2. Set an environment variable named **AzureWebJobsStorage** set to a connection string e.g. to the storage emulator.
+3. Set an environment variable named **AzureWebJobsDashboard** with the same value as **AzureWebJobsStorage**.
+4. Run Azure Storage Emulator 5.2.
+5. Run the unit tests via Visual Studio Test Explorer by selecting "Run All"
+
+## Change flow
```

Comment:
I think the order of these last three sections should be reversed. In other words, the first thing we should explain is the overall process. The next thing to explain is how to run the automated tests. Lastly we should describe how to test the changes as a NuGet package.

Reply:
Shall we have the following order:

1. Change flow
2. Running the tests
3. Testing code changes

Or you prefer that we have new dedicated section describing the overall process?

Reply:
I'm fine with how you reordered it in your latest commit. This is a good starting point that covers all the fundamentals. After we have this in place, I'd like to spend more time studying other open source contribution guidelines to see whether/how we can improve the overall structure by providing additional context.

Reply:
Sounds very good.